packages/labs/ssr/README.md

```diff
@@ -51,7 +60,83 @@ const ssrResult = await (renderModule(
   [{some: "data"}]         // Arguments to function
 ) as Promise<Iterable<unknown>>);
 
-for (let text of ssrResult) {
-  console.log(text);
-}
+// ...
+
+context.body = Readable.from(ssrResult);
+```
+
+## Client usage
+
+### Hydrating lit templates
+
+In order to "hydrate" lit templates, which is the process of having lit re-associate the expressions of a lit template with which nodes they should update in the DOM, the `experimental-hydrate` module is provided in the `lit` package. Prior to updating a server-rendered container using `render`, you should first call `hydrate` on that container using the same template and data that was used to render on the server:
+
+```js
+import {myTemplate} from './my-template.js';
+import {render, hydrate} from `lit/experimental-hydrate.js`;
+
+// Initial hydration required before render:
+const initialData = /* should be same data used to render on the server */;
+hydrate(myTemplate(initialData), document.body);
+
+// After hydration, render will efficiently update the server-rendered DOM:
+const update = (data) => render(myTemplate(data), document.body);
 ```
+
+### Hydrating LitElements
+
+When `LitElement`s are server rendered, their shadow root contents are emitted inside a `<template shadowroot>`, also known as a [Declarative Shadow Root](https://web.dev/declarative-shadow-dom/), a new browesr feature that shipped in [Chrome 90](https://developer.chrome.com/blog/new-in-chrome-90/#declarative). Declarative shadow roots automatically attach their contents to a shadow root on the template's parent element when parsed. For browsers that do not yet implement declarative shadow root, there is a [`template-shadowroot`](https://github.com/webcomponents/template-shadowroot) ponyfill, described below.
+
+Because the `hydrate` function above does not descend into shadow roots, it only works on one scope of the DOM at a time. To hydrate `LitElement` shadow roots, load the `lit/hydrate-support.js` module, which installs support for `LitElement` automatically hydrating itself when it detects it was server-rendered with declarative shadow DOM. This module should be loaded before the `lit` module is loaded, to ensure hydration support is properly installed.
+
+Put together, an HTML page that was server rendered and containing `LitElement`s in the main document might look like this:
+
+```js
+import {render} from '@lit-labs/ssr/lib/render-global.js';
+import './app-components.js';
+
+const ssrResult = render(html`
```

Comment:
I think it's fine for now. In a _typical_ WC-based app, custom elements are the top-level template owners for dynamic content on the page, not _another_ top level non-component actor, so on the server the top-level template is static (or at least, not dynamic on the client; it might use data to render out meta tags or something that won't change on the client). 

I get that this poses a conundrum for e.g. light-dom server rendering because in even when using `renderLight` and the data is coming from a highlander custom element as in `<x-app>${renderLight()}</x-app>`, the directive is invoked by the outer template. But we can get into that once we flesh out those patterns.

Hydrating the top-level template (implying there is dynamic data on the client) also means we'd need to add the `defer-hydration` configurability, because in that case even top-level elements should be deferred. So let's plan on tackling that when we flesh out light-dom rendering examples.

I can add a note for now indicating the outer template in the example is not dynamic and need not be hydrated on the client, but that there are other patterns.

Reply:
A note's fine for now

Reply:
Added note